examples/datagen_with_al/README.md

```diff
@@ -0,0 +1,221 @@
+    --qg_output_dir tmp \
+    --qg_model_name models/qg/bart-large_squad/checkpoint-12000 \
+    --qg_per_device_train_batch_size 1 \
+    --qg_gradient_accumulation_steps 24 \
+    --qg_save_total_limit 1 \
+    --qg_metric_for_best_model loss \
+    --qg_max_steps 500 \
+    --qg_per_device_eval_batch_size 5 \
+    --qg_learning_rate 3e-5 \
+    --qg_eval_steps 10 \
+    --qg_logging_steps 10 \
+    --qg_save_steps 10
+```
+> Make sure to pick the best checkpoint from previous training for `--qg_model_name`, i.e. the one with the lowest number of steps.
+> Also we're setting `--qg_max_steps` since dataset size is small and `--num_train_epochs` would result in too few training steps (if not set to a huge value)
+
+### Generate data
+
+In this step, we will generate question-answer pairs from unlabelled documents given a fine-tuned data generation model, e.g. after applying AL in the previous step:
+
+```bash
+python run.py \
+    --do_generate \
+    --predict_dataset pubmed-20 \
+    --num_workers 5 \
+    --rc_output_dir tmp \
+    --max_gen_length 300 \
+    --gen_output_path generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50 \
+    --qg_model_name models/qg/bart-large-squad_al-bioasq-dsp-4x50_round-4-of-4/checkpoint-500 \
+    --qg_output_dir tmp \
+    --qg_per_device_eval_batch_size 5
+```
+> Make sure to pick the best checkpoint from last iteration of AL for `--qg_model_name`, i.e. the one with the lowest number of steps.
+> 
+
+By default, the generated data is stored as a `datasets:Dataset` to `./generated_data`. This can be changed using `--gen_output_path`.
+> Note that you can use sharding to split a large set of documents into several runs by specifying `--shard_size` or `--num_shards`. The shard indices are then computed deterministically and the zero-based shard indices to run can optionally be set using `--shards`. Generated data is then stored in subfolders of the output directory named with the indices of the shards.
+
+### Filter generated data
+
+Since generated data is noisy, we will filter them in a next step.
+The script `filter_gen_data.py` does this:
+```bash
+python filter_gen_data.py <paths to generated data> <output dir> --filter {lm,rt}
+```
+> Note that in case of sharding applied to the data generation, you will have to give all shards to the script, for example `generated_data/*`.
+
+There are two filtering methods integrated: One based on LM score and the other using an RC model.
+
+#### LM score filtering
+
+For applying filtering using the LM score:
+```bash
+python filter_gen_data.py generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50 generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50_lm --filter lm
+```
+
+#### RTcons filtering
+
+For applying filtering using an RC model we need to train an RC model first.
+We suggest fine-tuning a model on SQuAD first followed by fine-tuning on the selected, annotated data:
+```bash
+# fine-tune RC model on SQuAD
+python run.py \
```

Comment:
can we use run_mrc from primeqa to fine-tune model on squad?

Reply:
Yes that works as well, but currently the MRC functionality is needed for AL anyway (if used with an RC model).